internal/notifier/client.go

```diff
@@ -32,29 +32,117 @@ import (
 	"github.com/hashicorp/go-retryablehttp"
 )
 
-type requestOptFunc func(*retryablehttp.Request)
+type postOptions struct {
+	proxy             string
+	certPool          *x509.CertPool
+	requestModifier   func(*retryablehttp.Request)
+	responseValidator func(*http.Response) error
+}
+
+type postOption func(*postOptions)
+
+func postMessage(ctx context.Context, address string, payload interface{}, opts ...postOption) error {
+	options := &postOptions{
+		// Default validateResponse function verifies that the response status code is 200, 202 or 201.
+		responseValidator: func(resp *http.Response) error {
+			if resp.StatusCode == http.StatusOK ||
+				resp.StatusCode == http.StatusAccepted ||
+				resp.StatusCode == http.StatusCreated {
+				return nil
+			}
+
+			return fmt.Errorf("request failed with status %s", resp.Status)
```

Comment:
In the previous version of the default validation we were also reading the body and returning it in the error.

Reply:
Actually the body was reported in the error handling around L93, but it was not clear.
I refactored error handling in 900ea8cfdfb35a1dd183da2235eb7109aaff3867